compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -1057,29 +1070,31 @@ impl<'a> MethodDef<'a> {
 
         if !is_packed {
             let selflike_fields =
-                trait_.create_struct_field_access_fields(cx, selflike_args, struct_def);
+                trait_.create_struct_field_access_fields(cx, selflike_args, struct_def, false);
+            mk_body(cx, selflike_fields)
+        } else if always_copy {
+            let selflike_fields =
+                trait_.create_struct_field_access_fields(cx, selflike_args, struct_def, true);
             mk_body(cx, selflike_fields)
         } else {
+            // Neither packed nor copy. Need to use ref patterns.
```

Comment:
This seems to be a typo? We only get here if `is_packed && !always_copy`.

Reply:
Good catch. Fixed in #104391.